### ISSUE TITLE
Multiple items in order params doesn't seem to work.

### ISSUE THREAD

**Issue by tdestro** [stripe/stripe-go#251]
I think I may have found a bug. This seems to only work for me if I have one item. Also I noticed the tests in stripe-go/order/client_test.go in  func TestOrder(t *testing.T) don't test for orders with multiple items. I get this in the stripe panel:
<img width="744" alt="screen shot 2016-05-20 at 10 32 55 am" src="https://cloud.githubusercontent.com/assets/10113013/15431079/48a090be-1e76-11e6-8eb0-6c5b3a71ae32.png">

```

	order, err := sc.Orders.New(&stripe.OrderParams{
		Currency: "usd",
		Items: []*stripe.OrderItemParams{
			{
				Type: "sku",
				Parent: "sku_8KuzQIEccDprsR",
			},
			{
				Type: "sku",
				Parent: "sku_8KbgQAaeg7C2S9",
			},
		},
		Shipping: &stripe.ShippingParams{
			Name: "Jenny Rosen",
			Address: &stripe.AddressParams{
				Line1:      "1234 Main Street",
				City:       "Anytown",
				Country:    "US",
				PostalCode: "123456",
			},
			Phone: "6504244242",
		},
		Email: "jenny@ros.en",
		Params: stripe.Params{
			Meta: map[string]string{
				"foo": "bar",
			},
		},
	})

```

**Comment by brandur** [comment]
@tdestro Thanks for reporting!

And shoot — this is a really bad one. The `New` function is doing the right thing:

``` go
		if len(params.Items) > 0 {
			for i, item := range params.Items {
				body.Add("items[][description]", item.Description)
				body.Add("items[][type]", string(item.Type))
				body.Add("items[][amount]", strconv.FormatInt(item.Amount, 10))
...
```

But because we're using "Rack-y" form encoding which requires that arrays be specified as ordered sets of parameters using `arr[]` and `url.Values` is really just a Go map that doesn't guarantee ordering by order of insertion:

``` go
type Values map[string][]string
```

Then when the form is encoded, it comes out looking like this (which is nonsense):

```
currency=usd&email=jenny%40ros.en&email=jenny%40ros.en&items[][amount]=0&items[][amount]=0&items[][description]=&items[][description]=&items[][parent]=sku_8KuzQIEccDprsR&items[][parent]=sku_8KbgQAaeg7C2S9&items[][type]=sku&items[][type]=sku&metadata[foo]=bar&shipping[address][city]=Anytown&shipping[address][country]=US&shipping[address][line1]=1234+Main+Street&shipping[address][postal_code]=123456&shipping[name]=Jenny+Rosen&shipping[phone]=6504244242
```

I think we may need to rework the implementation entirely to get a workable fix.

**Comment by brandur** [comment]
And separately, the server side error message could stand to be improved. I believe it's erroring because an empty SKU was passed in and is showing that "empty string cannot be found" when it should read something more like "no SKU provided".

**Comment by brandur** [comment]
I've added an initial attempt at a fix for this in #252. Comments welcome!
